### PR TITLE
feat(DevTools): Focus follower

### LINKED ISSUE #12241: [DevTools] Focus follower
**Is your feature request related to a problem? Please describe.**
It's not always easy to identify the active control when using the DevTools

**Describe the solution you'd like**
Added an option to the DevTools that allows you to decorate the active control so that it is clearly visible.

**Describe alternatives you've considered**
none

**Additional context**
none

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

Added Focus Highlighter item to DevTools Overlays menu, which allows you to highlight the currently focused item with the specified color.

feature

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues

Fixes #12241

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0039273-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0039624-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

File: src/Avalonia.Diagnostics/Diagnostics/ViewModels/MainViewModel.cs [hunk 1]
```diff
@@ -271,7 +276,16 @@ public void Dispose()
...
         private void UpdateFocusedControl()
         {
-            FocusedControl = KeyboardDevice.Instance?.FocusedElement?.GetType().Name;
+            var element = KeyboardDevice.Instance?.FocusedElement;
+            FocusedControl = element?.GetType().Name;
+            _currentFocusHighlightAdorner?.Dispose();
+            if (FocusHighlighter is IBrush brush 
+                && element is InputElement input 
+                && TopLevel.GetTopLevel(input) is { } topLevel 
+                && topLevel.GetType().FullName != "Avalonia.Diagnostics.Views.MainWindow")
```
Comment (maxkatz6) [inline src/Avalonia.Diagnostics/Diagnostics/ViewModels/MainViewModel.cs:285]: In this project, you can simply check with "is Avalonia.Diagnostics.Views.MainWindow". Why do you need reflection?

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0039646-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Comment by maxkatz6** [comment]
While this PR looks good to me, we can try to reuse normal FocusAdorner.
I.e. each control has already integrated focus adorner functionality, it's just not always visible on focus.

As a possible implementation, I am thinking about adding this toggle property to AdornerLayer class. Which can be read on this condition https://github.com/AvaloniaUI/Avalonia/blob/ae765e6022d4f097dd4b5dd6612f7210aa867e89/src/Avalonia.Controls/Control.cs#L406-L407

The same DevTools toggle can also override AdornerLayer.DefaultFocusAdorner property to make it more visible as well.

What do you think?

**Comment by workgroupengineering** [comment]
> While this PR looks good to me, we can try to reuse normal FocusAdorner. I.e. each control has already integrated focus adorner functionality, it's just not always visible on focus.
> 
> As a possible implementation, I am thinking about adding this toggle property to AdornerLayer class. Which can be read on this condition
> 
> https://github.com/AvaloniaUI/Avalonia/blob/ae765e6022d4f097dd4b5dd6612f7210aa867e89/src/Avalonia.Controls/Control.cs#L406-L407
> 
> The same DevTools toggle can also override AdornerLayer.DefaultFocusAdorner property to make it more visible as well.
> 
> What do you think? My bad that I thought about it only now.

I'll try.

**Comment by workgroupengineering** [comment]
@maxkatz6  I tried to apply your suggestion in commit 3d60326ec1212038dd3f39850b34d5938d052cb7, I don't like how solution increases complexity and there are some problems with popup windows. Do I continue on the path you suggested?

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0039794-beta`. (feed url: https://nuget-feed-all.avaloniaui.net/v3/index.json) [PRBUILDID]

**Pull request merged**